python_modules/libraries/dagster-dlt/dagster_dlt/components/dlt_load_collection/component.py

```diff
@@ -129,15 +134,19 @@ def build_defs(self, context: ComponentLoadContext) -> dg.Definitions:
                 name=f"dlt_assets_{load.source.name}_{load.pipeline.dataset_name}",
                 dagster_dlt_translator=load.translator,
             )
-            def dlt_assets_def(context: AssetExecutionContext):
-                yield from self.execute(context, load.source, load.pipeline)
+            def dlt_assets_def(
+                context: AssetExecutionContext, dlt_pipeline_resource: DagsterDltResource
+            ):
+                yield from self.execute(context, dlt_pipeline_resource)
 
             output.append(dlt_assets_def)
 
-        return dg.Definitions(assets=output)
+        return dg.Definitions(
+            assets=output, resources={"dlt_pipeline_resource": DagsterDltResource()}
```

Comment:
i think we should not use `resources` here, lest it be impossible to have multiple dlt components in your project at once.

the alternative (given that this resource is not configured at all) is to do what the DbtProjectComponent does and make the resource a property of the component